File: Backend/src/main/java/com/smartagriculture/service/FarmerService.java
package com.smartagriculture.service;

import com.smartagriculture.dto.farmer.FarmerRequest;
import com.smartagriculture.dto.farmer.FarmerResponse;
import java.util.List;

public interface FarmerService {

    FarmerResponse create(FarmerRequest request);

    FarmerResponse update(Long id, FarmerRequest request);

    void delete(Long id);

    FarmerResponse getById(Long id);

    List<FarmerResponse> findAll();
}

File: Frontend/src/services/farmerService.ts
import api from './api';
import type { FarmerRequest, FarmerResponse } from '../types';

export const farmerService = {
  list: () => api.get<FarmerResponse[]>('/farmers'),
  getById: (id: number) => api.get<FarmerResponse>(`/farmers/${id}`),
  create: (data: FarmerRequest) => api.post<FarmerResponse>('/farmers', data),
  update: (id: number, data: FarmerRequest) => api.put<FarmerResponse>(`/farmers/${id}`, data),
  delete: (id: number) => api.delete(`/farmers/${id}`),
};
